test: wiki-absent status tests are not hermetic — real ~/.memtomem-wiki leaks through the import-time-frozen DEFAULT_WIKI_PATH

## Symptom

Three wiki-absent degradation tests fail on any machine that has a real `~/.memtomem-wiki` (i.e. any dev box that has used the ADR-0008 wiki layer), while passing on CI runners (no wiki present):

- `tests/test_context_status.py::test_collect_wiki_absent_degrades_to_stale_pin`
- `tests/test_cli_status_all_projects.py::test_wiki_absent_degradation_note`
- `tests/test_web_routes_context_status_all.py::test_wiki_absent_degrades_to_stale_pin_rows`

Typical failure — the real wiki's HEAD leaks into the "absent" scenario:

```
assert entry["wiki_head"] is None
E       AssertionError: assert '0fad5dee21fbf60cabe65e5bbe229c964b6adef5' is None
```

Reproduced on a clean `origin/main` checkout (`cd0ae767`), so unrelated to any in-flight branch.

## Root cause (verified, two interacting parts)

1. **`DEFAULT_WIKI_PATH` is frozen at import time.** `wiki/store.py:24` evaluates `Path.home() / ".memtomem-wiki"` when the module is imported — i.e. under the *real* `HOME` during pytest collection. The tests' `set_home(monkeypatch, home)` sandbox (HOME + USERPROFILE) is applied *after* import, so it never affects the constant:

   ```python
   >>> from memtomem.wiki.store import _wiki_path_from_env
   >>> os.environ["HOME"] = "/tmp/sandbox-home"; os.environ.pop("MEMTOMEM_WIKI_PATH", None)
   >>> Path.home()
   PosixPath('/tmp/sandbox-home')
   >>> _wiki_path_from_env()
   PosixPath('/Users/<user>/.memtomem-wiki')   # frozen to the import-time HOME
   ```

2. **The hermeticity helpers guarantee the fallback.** All three tests sandbox via a shared-shape helper (`test_context_status.py:1128` `_collect_home`, and its siblings at `test_cli_status_all_projects.py:55`, `test_web_routes_context_status_all.py:57`) that does `monkeypatch.delenv("MEMTOMEM_WIKI_PATH")` — intended to shield the test from a developer's env override, but combined with (1) it *removes the only lever that works* and pins resolution to the frozen real-home constant. Exporting `MEMTOMEM_WIKI_PATH=/nonexistent` before pytest does not help for exactly this reason (verified: still 3 failed).

Wiki-*seeding* tests are unaffected because the shared `wiki_root` fixture (`tests/_wiki_fixtures.py:36`) uses `monkeypatch.setenv("MEMTOMEM_WIKI_PATH", ...)`, which takes the env branch of `_wiki_path_from_env()` and never touches the constant. Only the wiki-absent tests rely on the delenv+`set_home` path.

## Fix options

- **A (leaning) — make the fallback lazy in `_wiki_path_from_env()`** (`wiki/store.py:117-121`): return `Path.home() / ".memtomem-wiki"` computed at call time instead of the import-time constant. One-line source fix; `set_home` then works as designed and all three tests pass unmodified (their delenv becomes genuinely correct hermeticity). Decide what to do with the exported `DEFAULT_WIKI_PATH` (`wiki/__init__.py` `__all__`): keep as documentation-only default (note the freeze caveat in its docstring) or replace with a `default_wiki_path()` helper; `tests/test_wiki_store.py:25` pins the constant's value and `:38` (`test_at_default_falls_back_to_home`) already asserts the lazy-equivalent behavior, so both survive option A with the constant kept.
- **B — test-only:** switch the three helpers from `delenv` to `setenv("MEMTOMEM_WIKI_PATH", str(<sandbox>/wiki-absent))`. Smaller blast radius, but leaves the import-time freeze as a trap for every future `set_home`-based wiki test (this is the second local-only-failure class this trap has produced; the first required the `wiki_root` fixture's setenv).

Either way, ≤3 files. A regression pin should run one wiki-absent test with a fabricated real-home wiki (tmp `HOME` at import is not controllable — so pin via the lazy resolver directly: monkeypatch HOME after import and assert `_wiki_path_from_env()` follows it).

## Environment

Locally observed on macOS (Darwin 25.5.0), Python 3.12, `origin/main @ cd0ae767`; CI green on the same commit because runners have no `~/.memtomem-wiki`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/packages/memtomem/src/memtomem/wiki/store.py
+++ b/packages/memtomem/src/memtomem/wiki/store.py
@@ -22,7 +22,13 @@ from pathlib import Path, PurePosixPath
 logger = logging.getLogger(__name__)
 
 DEFAULT_WIKI_PATH: Path = Path.home() / ".memtomem-wiki"
-"""Default wiki location — overridable via ``MEMTOMEM_WIKI_PATH`` env."""
+"""Default wiki location — overridable via ``MEMTOMEM_WIKI_PATH`` env.
+
+Documentation-only: frozen under the importing process's home at import
+time. Runtime resolution goes through :func:`_wiki_path_from_env`, which
+re-reads ``Path.home()`` at call time so a ``HOME`` override applied after
+import (e.g. a test sandbox) is honored (#1506).
+"""
 
 WIKI_ASSET_TYPES: tuple[str, ...] = ("skills", "agents", "commands")
 """Asset directory names at the wiki root. Order is significant for listing."""
@@ -118,7 +124,9 @@ def _wiki_path_from_env() -> Path:
     env = os.environ.get("MEMTOMEM_WIKI_PATH")
     if env:
         return Path(env).expanduser()
-    return DEFAULT_WIKI_PATH
+    # Call-time Path.home(), not DEFAULT_WIKI_PATH — the constant is frozen
+    # at import and would ignore a HOME override applied afterwards (#1506).
+    return Path.home() / ".memtomem-wiki"
 
 
 # Match the ``userinfo@`` segment of a ``scheme://userinfo@host`` URL. The

--- a/packages/memtomem/tests/test_wiki_store.py
+++ b/packages/memtomem/tests/test_wiki_store.py
@@ -17,7 +17,10 @@ from memtomem.wiki.store import (
     WikiNotFoundError,
     WikiNothingToCommitError,
     WikiStore,
+    _wiki_path_from_env,
 )
+
+from .helpers import set_home
 
 
 class TestDefaultPath:
@@ -36,6 +39,19 @@ class TestDefaultPath:
         monkeypatch.delenv("MEMTOMEM_WIKI_PATH", raising=False)
         store = WikiStore.at_default()
         assert store.root == Path.home() / ".memtomem-wiki"
+
+    def test_fallback_follows_home_set_after_import(
+        self, monkeypatch: pytest.MonkeyPatch, tmp_path: Path
+    ) -> None:
+        """Regression pin for #1506 — the fallback must re-read ``Path.home()``
+        at call time. The import-time-frozen ``DEFAULT_WIKI_PATH`` made
+        wiki-absent tests leak the real ``~/.memtomem-wiki`` on dev machines
+        despite their ``set_home`` sandbox.
+        """
+        monkeypatch.delenv("MEMTOMEM_WIKI_PATH", raising=False)
+        sandbox_home = tmp_path / "sandbox-home"
+        set_home(monkeypatch, sandbox_home)
+        assert _wiki_path_from_env() == sandbox_home / ".memtomem-wiki"
 
 
 class TestInitScratch:

